content/essentials/stores.mdx

```diff
@@ -0,0 +1,203 @@
+
+```jsx
+setStore("users", (otherUsers) =>[ 
+    ...otherUsers,
+    {
+    id: 3,
+    username: 'michael584',
+    location: 'Nigeria',
+    loggedIn: false
+}])
+```
+
+Modifying a store using this method is referred to as "path syntax." 
+In this approach, the initial arguments are used to specify the keys that lead to the target value you want to modify, while the last argument provides the new value.
+
+<Aside>
+  Separating read and write capabilities to a store presents a valuable debugging advantage.
+  It facilitates the tracking and control of which components can access or change store values.
+</Aside>
+
+### Path syntax flexibilty 
+
+String keys are frequently used to precisely target particular values.
+By specifying these exact key names, you can direclty retrieve the desired information.
+However, path syntax goes beyond string keys and offers more versatility when accessing targeted values.
+
+Instead of employing the use of just single string keys, there is the option of using an array of keys.
+This method grants you the ability to select multiple properties within the store, facilitating accessed to nested structures.
+Alternatively, you can use filtering functios to access keys based on dynamic conditions or specific rules.
+
+[View on Eraser![](https://app.eraser.io/workspace/maDvFw5OryuPJOwSLyK9/preview?elements=aUOwffJd4QDWWDi22XI8DA&type=embed)](https://app.eraser.io/workspace/maDvFw5OryuPJOwSLyK9?elements=aUOwffJd4QDWWDi22XI8DA)
+
+The flexibility in path syntax makes for efficient navigation and data retrieval from your store, regardless of the store's complexity or the requirement for dynamic access scenarios within your application.
+
+## Store utilities
+
+### Store updates with `produce`
+ 
+Rather than directly modifying a store with setters, Solid has the `produce` utility.
+This utility provides a way to work with data as if it were a [mutable](https://developer.mozilla.org/en-US/docs/Glossary/Mutable) JavaScript object.
+`produce` also provides a way to make changes to multiple properties at the same time which eliminates the need for multiple setter calls.
+
+```jsx
+import { produce } from "solid-js/store"
+
+// without produce
+setStore(0, 'username', 'newUsername')
+setStore(0, 'location', 'newLocation')
+
+// with produce
+setStore(0,
+produce((user) => {
+  user.username = 'newUsername'
+  user.location = 'newLocation'
+}))
+```
+
+`produce` and `setState` do have distinct functionalities. 
+While both can be used to modify the state, the key distinction lies in how they handle data.
+`produce` allows you to work with a temporary draft of the state, apply the changes, then produce a new [immutable](https://developer.mozilla.org/en-US/docs/Glossary/Immutable) version it.
+Comparatively, `setState` provides a more straightforward way of updating the state directly, without creating a new version.
```

Comment:
"Comparatively, `setStore` provides a more straightforward way of updating the store directly, without creating a new version."

Please remember context